server/src/main/java/org/opensearch/action/admin/indices/tiering/TieringUtils.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.indices.tiering;
+
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.index.Index;
+import org.opensearch.index.IndexModule;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static org.opensearch.index.IndexModule.INDEX_TIERING_STATE;
+
+/**
+ * Utility class for tiering operations
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class TieringUtils {
+
+    /**
+     * Constructs a HotToWarmTieringResponse from the rejected indices map
+     *
+     * @param rejectedIndices the rejected indices map
+     * @return the HotToWarmTieringResponse object
+     */
+    public static HotToWarmTieringResponse constructToHotToWarmTieringResponse(final Map<Index, String> rejectedIndices) {
+        final List<HotToWarmTieringResponse.IndexResult> indicesResult = new LinkedList<>();
+        for (Map.Entry<Index, String> rejectedIndex : rejectedIndices.entrySet()) {
+            indicesResult.add(new HotToWarmTieringResponse.IndexResult(rejectedIndex.getKey().getName(), rejectedIndex.getValue()));
+        }
+        return new HotToWarmTieringResponse(true, indicesResult);
+    }
+
+    /**
+     * Checks if the specified index is in the "hot" tiering state.
+     *
+     * @param indexMetadata the metadata of the index
+     * @return true if the index is in the "hot" tiering state, false otherwise
+     */
+    public static boolean isHotIndex(final IndexMetadata indexMetadata) {
+        return isIndexWithTier(indexMetadata, IndexModule.TieringState.HOT);
+    }
+
+    /**
+     * Checks if the specified index is in the specified tiering state.
+     *
+     * @param indexMetadata the metadata of the index
+     * @param tieringState  the tiering state to check for
+     * @return true if the index is in the specified tiering state, false otherwise
+     */
+    public static boolean isIndexWithTier(final IndexMetadata indexMetadata, IndexModule.TieringState tieringState) {
+        return tieringState.name().equals(INDEX_TIERING_STATE.get(indexMetadata.getSettings()));
+    }
```

Comment:
These 2 seems to fit well in IndexMetadata class itself as non-static method. Also lets rename `isIndexWithTier` to `isIndexInTier`

Reply:
updated